app/v2/notifications/notification_schemas.py

```diff
@@ -192,40 +192,89 @@
-    return {"id": notification.id,
-            "reference": notification.client_reference,
-            "content": {'body': body,
-                        'from_number': from_number},
-            "uri": "{}v2/notifications/{}".format(url_root, str(notification.id)),
-            "template": __create_template_from_notification(notification=notification,
-                                                            url_root=url_root,
-                                                            service_id=service_id),
-            "scheduled_for": scheduled_for if scheduled_for else None
-            }
+post_letter_request = {
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "description": "POST letter notification schema",
+    "type": "object",
+    "title": "POST v2/notifications/letter",
+    "properties": {
+        "reference": {"type": "string"},
+        "template_id": uuid,
+        "personalisation": letter_personalisation
+    },
+    "required": ["template_id", "personalisation"]
+}
+
+letter_content = {
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "description": "Letter content for POST letter notification",
+    "type": "object",
+    "title": "notification letter content",
+    "properties": {
+        "body": {"type": "string"},
+        "subject": {"type": "string"}
+    },
+    "required": ["body", "subject"]
+}
+
+post_letter_response = {
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "description": "POST sms notification response schema",
+    "type": "object",
+    "title": "response v2/notifications/letter",
+    "properties": {
+        "id": uuid,
+        "reference": {"type": ["string", "null"]},
+        "content": letter_content,
+        "uri": {"type": "string", "format": "uri"},
+        "template": template,
+        "scheduled_for": {"type": ["string", "null"]}
+    },
+    "required": ["id", "content", "uri", "template"]
+}
+
+
+def create_post_sms_response_from_notification(notification, content, from_number, url_root, scheduled_for):
+    noti = __create_notification_response(notification, url_root, scheduled_for)
+    noti['content'] = {
+        'from_number': from_number,
+        'body': content
+    }
+    return noti
+
+
+def create_post_email_response_from_notification(notification, content, subject, email_from, url_root, scheduled_for):
+    noti = __create_notification_response(notification, url_root, scheduled_for)
+    noti['content'] = {
+        "from_email": email_from,
+        "body": content,
+        "subject": subject
+    }
+    return noti
+
+
+def create_post_letter_response_from_notification(notification, content, subject, url_root, scheduled_for):
+    noti = __create_notification_response(notification, url_root, scheduled_for)
+    noti['content'] = {
+        "body": content,
+        "subject": subject
+    }
+    return noti
 
 
```

Comment:
Would be nice to put these functions somewhere else, away from where we define schemas?